# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1213379716 14400
#      Fri Jun 13 13:55:16 2008 -0400
# Node ID 76142609e8d2002346cf715a15295e98a789a55c
# Parent  c123bd09172e587461bb7e51a594c64f5e618062
Remove some float-functions ambiguities.

diff --git a/liboctave/ChangeLog b/liboctave/ChangeLog
--- a/liboctave/ChangeLog
+++ b/liboctave/ChangeLog
@@ -1,8 +1,13 @@
+2008-06-13  Michael Goffioul  <michael.goffioul@gmail.com>
+
+	* lo-mappers.cc (arg): Remove ambiguity about atan2(float,float) usage.
+	* fCmplxDET.cc (FloatComplexDET::initialize10): Likewise, for pow.
+
 2008-06-11  John W. Eaton  <jwe@octave.org>
 
 	* so-array.h, so-array.cc, Array-so.cc: Delete.
 	* Makefile.in: Remove them from the lists.
 
 2008-06-05  John W. Eaton  <jwe@octave.org>
 
 	* oct-shlib.cc (octave_base_shlib::remove): Only dereference
diff --git a/liboctave/fCmplxDET.cc b/liboctave/fCmplxDET.cc
--- a/liboctave/fCmplxDET.cc
+++ b/liboctave/fCmplxDET.cc
@@ -52,17 +52,17 @@ FloatComplexDET::value_will_underflow (v
 void
 FloatComplexDET::initialize10 (void)
 {
   if (c2 != static_cast<float> (0.0))
     {
       float etmp = e2 / xlog2 (static_cast<float>(10));
       e10 = static_cast<int> (xround (etmp));
       etmp -= e10;
-      c10 = c2 * static_cast<float> (pow (10.0, etmp));
+      c10 = c2 * static_cast<float> (pow (10.0f, etmp));
     }
 }
 
 void
 FloatComplexDET::initialize2 (void)
 {
   if (c10 != static_cast<float> (0.0))
     {
diff --git a/liboctave/lo-mappers.cc b/liboctave/lo-mappers.cc
--- a/liboctave/lo-mappers.cc
+++ b/liboctave/lo-mappers.cc
@@ -397,17 +397,17 @@ xmax (const Complex& x, const Complex& y
 }
 
 
 // float -> float mappers.
 
 float
 arg (float x)
 {
-  return atan2 (0.0, x);
+  return atan2 (0.0f, x);
 }
 
 float
 conj (float x)
 {
   return x;
 }
 
